### PR TITLE
CTED-1398 added a forum loader

### PR DESCRIPTION
**Pull request by PPrinsloo**

### CODE REVIEW

File: amd/src/mod_hsuforum_loader.js [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+            $('.article :input').attr('disabled', true);
+            $('.hsuforum-menu :input').attr('disabled', true);
+
+            waitForElement("body", 30000).then(function() {
+                $('.article :input').removeAttr('disabled');
+                $('.hsuforum-menu :input').removeAttr('disabled');
+                $('.article').show();
+                $('#loading-container').hide();
+            }).catch(() => {
+                alert("element did not load in 30 seconds");
```
Comment (JJSwanevelder) [inline amd/src/mod_hsuforum_loader.js:39]: Rather use throw instead of alert

File: discuss.php [hunk 2]
```diff
@@ -223,6 +223,8 @@
     $PAGE->set_heading($course->fullname);
     echo $OUTPUT->header();
 
+echo $OUTPUT->render_from_template('mod_hsuforum/loading', []);
```
Comment (JJSwanevelder) [inline discuss.php:226]: Indent missing

File: amd/src/mod_hsuforum_loader.js [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+        });
+    }
+
+
+    return {
+        init: function () {
+            $('.article :input').attr('disabled', true);
+            $('.hsuforum-menu :input').attr('disabled', true);
+
+            waitForElement("body", 30000).then(function() {
```
Comment (JJSwanevelder) [inline amd/src/mod_hsuforum_loader.js:33]: Is this the failsafe to remove after x amount if not removed already?

**Review by JJSwanevelder — CHANGES_REQUESTED** [review]
Some minor comments - nice work on promise :)

File: amd/src/mod_hsuforum_loader.js (hunk 1 above)
Comment (kurvinh) [reply amd/src/mod_hsuforum_loader.js:39]: agreed. don't want a user to see that.

File: amd/src/mod_hsuforum_loader.js (hunk 3 above)
Comment (PPrinsloo) [reply amd/src/mod_hsuforum_loader.js:33]: yes

**Review by JJSwanevelder — APPROVED** [review]
👍

**Review by kurvinh — APPROVED** [review]
👍

**Pull request merged**